src/main/java/com/github/cameltooling/lsp/internal/parser/CamelKModelineInsertionParser.java

```diff
@@ -0,0 +1,62 @@
+package com.github.cameltooling.lsp.internal.parser;
+
+import com.github.cameltooling.lsp.internal.completion.modeline.CamelKModelineInsertionProcessor;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.TextDocumentItem;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.Optional;
+
+public class CamelKModelineInsertionParser {
+
+    public boolean canPutCamelKModeline(Position position, TextDocumentItem textDocumentItem) {
+        int currentLine = position.getLine();
+
+        //Save the lines to not have to access multiple times to file?
+        return isCamelKFile(textDocumentItem)
+                && lineIsEmpty(currentLine, textDocumentItem)
+                && noModelineInsertedAlready(textDocumentItem)
+                && previousLinesAreCommentsOrEmpty(currentLine,textDocumentItem);
+    }
+
+    private boolean isCamelKFile(TextDocumentItem document) {
+        return FileType.getFileTypeCorrespondingToUri(document.getUri()).isPresent();
+    }
+
+    private boolean lineIsEmpty(int line, TextDocumentItem textDocumentItem) {
+        return new ParserFileHelperUtil().getLine(textDocumentItem, line).isEmpty();
+    }
+
+    private boolean noModelineInsertedAlready(TextDocumentItem textDocumentItem) {
+        //Will do later
```

Comment:
hard to guess who will do it later when we will read the code in 6 months.
Also priorities can change for one day to another so we never know.